### ISSUE TITLE
How to invoke tf.initialize_all_variables() in C++?

### ISSUE THREAD

**Issue by parilo** [tensorflow/tensorflow#856]
Hi, and thank you for your job)
I'm newbie in tensorflow.
Btw, how to execute ops in C++ such as tf.initialize_all_variables() or some ops returned by optimizers
self.train_op = self.optimizer.apply_gradients(gradients)
?

I've tried 
init_all_vars_op = tf.initialize_variables(tf.all_variables(), name='init_all_vars_op')
...
tf.train.write_graph(session.graph_def, 'models/', 'graph.pb', as_text=False)
in python and then in C++:

		status = session->Run(inputs, {"init_all_vars_op"}, {}, &outputs);
		if (!status.ok()) {
			std::cout << "tf error: " << status.ToString() << "\n";
			return;
		}

so I got

tf error: Invalid argument: FetchOutputs init_all_vars_op: output index too large, must be < 0

here is stackowerflow question
http://stackoverflow.com/questions/34975884/how-to-invoke-tf-initialize-all-variables-in-c-tensorflow

**Comment by josh11b** [comment]
Looks like this is an unclear error message.  You are asking for the output of the "init_all_vars_op", but it doesn't have any output.  You just want to run that, which you do by supplying it to the third parameter to Run(), see https://www.tensorflow.org/versions/master/api_docs/cc/ClassSession.html#virtual_Status_tensorflow_Session_Run

status = session->Run(inputs, {}, {"init_all_vars_op"}, &outputs);
